Dynamic imports (#244)
javascript-tutorial/it.javascript.info: 1-js/13-modules/03-modules-dynamic-imports/article.md

# Dynamic imports

> Le istruzioni di export ed import che abbiamo visto nei capitolo precedenti sono detti "statici". La sintassi è veramente semplice.

Reviewer: ```suggestion
Le istruzioni di export ed import che abbiamo visto nei capitolo precedente sono detti "statici". La sintassi è molto semplice e rigorosa.
```

> Le istruzioni di export ed importa che abbiamo visto nei capitolo precedenti, sono detti "statici". La sintassi è veramente semplice.

Reviewer 2: ```suggestion
Le istruzioni di export ed import che abbiamo visto nei capitolo precedenti sono detti "statici". La sintassi è veramente semplice.
```

Come prima cosa, non possiamo generare dinamicamente parametri di `import`.

Il percorso al modulo deve essere una stringa, non può essere una chiamata a funzione. Questo non funzionerebbe:

```js
import ... from *!*getModuleName()*/!*; // Errore, sono ammesse solamente string
```

Secondo, non possiamo importare a run-time in base a determinate condizioni:

```js
if(...) {
  import ...; // Errore, non è possibile farlo!
}

{
  import ...; // Errore, non possiamo scrivere gli import all'interno di nessun blocco
}
```

> Questo accade perché `import`/`export` mirano a fornire uno scheletro per la struttura del codice. Questa è una buona cosa, poiché la struttura del codice può essere analizzata, i moduli possono essere raccolti and impacchettati in un singolo file (grazie ad alcuni strumenti), gli export inutilizzati possono essere rimossi ("tree-shaken"). Questo è possibile solamente perché la struttura degli imports/exports è semplice e pre-fissata.

Reviewer: ```suggestion
Questo accade perché `import`/`export` mirano a fornire uno scheletro per la struttura del codice. Questa è una buona cosa, poiché la struttura del codice può essere analizzata, i moduli possono essere raccolti and impacchettati in un singolo file (grazie ad alcuni strumenti) e gli export inutilizzati possono essere rimossi ("tree-shaken"). Questo è possibile solamente perché la struttura degli imports/exports è semplice e preimpostata.
```

> Ma come possiamo importare un modulo dinamicamente, al volo?

Reviewer 2: ```suggestion
Ma come possiamo importare un modulo dinamicamente, secondo necessità?
```

> Ma come possiamo importare un modulo dinamicamente, secondo necessità?

Reviewer: ```suggestion
Ma come possiamo importare un modulo dinamicamente, a seconda delle necessità?
```

> ## L'espressione di import()

Reviewer: ```suggestion
## L'espressione import()
```

> L'espressione di `import(module)` carica il modulo e ritorna una promise, che si risolve in un oggetto che contiene tutti gli export del modulo. Può essere quindi invocata in un qualsiasi punto del codice.

Reviewer: ```suggestion
L'espressione `import(module)` carica il modulo e ritorna una promise, che si risolve in un oggetto che contiene tutti gli export del modulo. Può essere quindi invocata in un qualsiasi punto del codice.
```

> Possiamo utilizzarlo dinamicamente ovunque, ad esempio:

Reviewer: ```suggestion
Possiamo utilizzarla dinamicamente ovunque, ad esempio:
```

}
```

> ...Allora il dyamic import, può essere scritto cosi:

Reviewer 2: ```suggestion
...Allora il dyamic import può essere scritto così:
```

}
```

> ...Poi, per potervi accedere, possiamo utilizzare la proprietà `default` dell'oggetto:

Reviewer: ```suggestion
...Quindi, per potervi accedere, possiamo utilizzare la proprietà `default` dell'oggetto:
```
